app/src/main/java/com/markkolenbrander/capstonenewsapp/ArticlesListFragment.kt

```diff
@@ -110,4 +132,41 @@ class ArticlesListFragment : Fragment() {
             swipe.isRefreshing = false
         }
     }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == R.id.app_bar_switch) {
+            viewModel.toggleNightMode()
+        }
+        return true
+    }
```

Comment:
You could replace this with a call to `onCreateOptionsMenu` instead:

```kotlin
override fun onCreateOptionsMenu(menu: Menu, menuInflater: MenuInflater) {
    menuInflater.inflate(R.menu.overflow_menu, menu)

    val switchView = menu
        .findItem(R.id.app_bar_switch_menu_item)
        .actionView
        ?.findViewById<SwitchCompat>(R.id.app_bar_switch)

    switchView?.isChecked = viewModel.darkThemeEnabled.value ?: false

    switchView?.setOnCheckedChangeListener { _, _ ->
        viewModel.toggleNightMode()
    }
}
```

You'd also need to modify the menu item id from `app_bar_switch` to `app_bar_switch_menu_item` and in your switch layout, provide the id `app_bar_switch` to the `SwitchCompat` element. 

Also make sure to call `setHasOptionsMenu(true)` in the `onCreateView()` method of this fragment, otherwise these menu callback functions won't fire.

Reply:
Note, the example above will toggle on/off dark mode, you could another preference bool in your PrefsStore (along with another "toggle" function) to trigger the "download over wifi only".  You'd then need to pass in the PrefsStore instance to the repository and do blocking fetch to determine whether or not you fetch data only wifi with:

```kotlin
runBlocking {
    prefsStore.downloadOverWifiOnly.first()
}
```